atomic_reactor/utils/cachi2.py

```diff
@@ -68,3 +68,49 @@ def convert_SBOM_to_ICM(sbom: Dict[str, Any]) -> Dict[str, Any]:
         {"purl": comp["purl"]} for comp in sbom["components"]  # type: ignore
     ]
     return icm
+
+
+def generate_request_json(
+    remote_source: Dict[str, Any], remote_source_sbom: Dict[str, Any],
+    remote_source_env_json: Dict[str, Dict[str, str]],
+) -> Dict[str, Any]:
+    """Generates Cachito like request.json
+
+    Cachito does provide request.json, for backward compatibility
+    as some tools are depending on it, we have to generate also request.json from cachi2
+    """
+    def gen_dependency(sbom_dep: Dict[str, Any]) -> Dict[str, str]:
+        """Generate a single dependency from SBOM"""
+        # we need to detect type from purl, this is just heuristics,
+        # we cannot reliably construct type from purl
+        heuristic_type = "unknown"
+        purl_type_map = {
+            "pkg:golang/": "go-package",
+            "pkg:npm/": "npm",
```

Comment:
You could also detect `gomod` by looking at the `type` qualifier in the purls
* `pkg:golang/github.com/cachito-testing/cachito-gomod-with-deps@v0.0.0-20201001160613-4c65d49cae6b?type=module`
* `pkg:golang/github.com/cachito-testing/cachito-gomod-with-deps@v0.0.0-20201001160613-4c65d49cae6b?type=package`

Reply:
updated